items/src/main/scala/weco/api/items/services/ItemUpdateService.scala

```diff
@@ -0,0 +1,62 @@
+package weco.api.items.services
+
+import grizzled.slf4j.Logging
+import weco.api.stacks.services.SierraService
+import weco.catalogue.internal_model.identifiers.{
+  IdState,
+  IdentifierType,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.locations.{
+  AccessCondition,
+  PhysicalLocation
+}
+import weco.catalogue.internal_model.work.{Item, Work, WorkState}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ItemUpdateService(
+  sierraService: SierraService
+)(implicit executionContext: ExecutionContext)
+    extends Logging {
+
+  private def isSierraId(sourceIdentifier: SourceIdentifier) =
+    sourceIdentifier.identifierType == IdentifierType.SierraSystemNumber
+
+  private def updateAccessCondition(
+    item: Item[IdState.Minted],
+    accessConditionOption: Option[AccessCondition]
+  ) =
+    item.locations.map {
+      case physicalLocation: PhysicalLocation =>
+        physicalLocation.copy(
+          accessConditions = accessConditionOption.toList
+        )
+      case location => location
+    }
+
+  private def refreshItem(srcId: SourceIdentifier, item: Item[IdState.Minted]) =
+    sierraService
+      .getAccessCondition(srcId)
+      .map {
+        case Right(accessConditionOption) =>
+          item.copy(
+            locations = updateAccessCondition(item, accessConditionOption)
+          )
```

Comment:
I'd be tempted to add another case to the pattern match, rather than passing around options:

```suggestion
        case Right(Some(accessCondition)) =>
          item.copy(
            locations = updateAccessCondition(item, accessCondition)
          )
        
        case Right(_) => item  
```

But I don't think it makes a big difference.